contracts/multisig/test/asset_proxy_owner.ts

```diff
@@ -1,511 +1,641 @@
+            const reg = createFunctionRegistration(1, 1, 1);
+            await assetProxyOwner.registerFunctionCallBypassWallet.awaitTransactionSuccessAsync(
+                true,
+                reg.functionSelectors[0],
+                reg.destinations[0],
+                reg.functionCallTimeLockSeconds[0],
             );
-            expect(isAssetProxyRegistered).to.equal(isRegistered);
+            const latestTimestamp = await getLatestBlockTimestampAsync();
+            const transactionConfirmationTime = new BigNumber(latestTimestamp);
+            const result = assetProxyOwner.assertValidFunctionCall.callAsync(
+                transactionConfirmationTime,
+                reg.functionSelectors[0],
+                reg.destinations[0],
+            );
+            expect(result).to.revertWith(RevertReason.CustomTimeLockIncomplete);
         });
-
-        it('should fail if registering a null address', async () => {
-            const addressToRegister = constants.NULL_ADDRESS;
-            const isRegistered = true;
-            const registerAssetProxyData = testAssetProxyOwner.registerAssetProxy.getABIEncodedTransactionData(
-                addressToRegister,
-                isRegistered,
-            );
-            const submitTxRes = await assetProxyOwnerWrapper.submitTransactionAsync(
-                testAssetProxyOwner.address,
-                registerAssetProxyData,
-                owners[0],
-            );
-            const log = submitTxRes.logs[0] as LogWithDecodedArgs<AssetProxyOwnerSubmissionEventArgs>;
-            const txId = log.args.transactionId;
-
-            await assetProxyOwnerWrapper.confirmTransactionAsync(txId, owners[1]);
-            await increaseTimeAndMineBlockAsync(SECONDS_TIME_LOCKED.toNumber());
-
-            const executeTxRes = await assetProxyOwnerWrapper.executeTransactionAsync(txId, owners[0]);
-            const failureLog = executeTxRes.logs[0] as LogWithDecodedArgs<AssetProxyOwnerExecutionFailureEventArgs>;
-            expect(failureLog.args.transactionId).to.be.bignumber.equal(txId);
-
-            const isAssetProxyRegistered = await testAssetProxyOwner.isAssetProxyRegistered.callAsync(
-                addressToRegister,
+        it('should be successful if an unregistered function is called after the default timelock', async () => {
+            const latestTimestamp = await getLatestBlockTimestampAsync();
+            const transactionConfirmationTime = new BigNumber(latestTimestamp).minus(DEFAULT_TIME_LOCK);
+            const reg = createFunctionRegistration(1, 1, 1);
+            const result = assetProxyOwner.assertValidFunctionCall.callAsync(
+                transactionConfirmationTime,
+                reg.functionSelectors[0],
+                reg.destinations[0],
+            );
+            expect(result).to.be.fulfilled('');
+        });
+        it('should be successful if a registered function is called after the custom timelock', async () => {
```

Comment:
Let's add a test for `should be successful if a registered function is called after the default timelock` to handle the case when `hasCustomTimelock == false`

Reply:
This would actually fail if the registered function had a timelock longer than the default.

Reply:
Added a test for that case.